projects/vdk-plugins/vdk-kerberos-auth/src/vdk/plugin/kerberos/minikerberos_authenticator.py

```diff
@@ -75,7 +75,12 @@ def _kinit(self) -> None:
             async def get_tgt():
                 await krb_client.get_TGT()
 
-            asyncio.run(get_tgt())
+            # Create a separate event loop to avoid interfering with users'
+            # data jobs utilizing asyncio, as well.
+            loop = asyncio.new_event_loop()
+            loop.run_until_complete(get_tgt())
+            loop.close()
```

Comment:
Can you please explain a bit more? I am not sure whether this is the correct root cause and the solution. But it's possible I am not understanding something.

asyncio.run (https://github.com/python/cpython/blob/main/Lib/asyncio/runners.py#L150) inside basically does this - create a new event loop and closes it. So I cannot see what changes here. It does a bit more robust cleaning but I would find it surprising if that is the issue.

Reply:
Updated the description. Hopefully, it is better worded now.
That being said, the issue with using `asyncio.run()` in our case is, that the run method sets the default event loop https://github.com/python/cpython/blob/3.10/Lib/asyncio/runners.py#L41, and when it closes it, other components that rely on it break.